### PR TITLE
Comment pairing

### PR DESCRIPTION
**Pull request by Sarahabidar**
create commentpopup component and render it in pairingitem

### CODE REVIEW

File: components/PairingItem.js [hunk 1]
```diff
@@ -1,40 +1,83 @@
...
+    if (commentToEdit) {
+      setEditingComment(commentToEdit);
+      setShowCommentPopup(true);
+    }
+  };
+
+  const handleDelete = (commentId) => {
+    const updatedComments = comments.filter((c) => c.id !== commentId);
+    setComments(updatedComments);
+  };
```
Comment (isabeen) [inline components/PairingItem.js]:
```suggestion
  const handleDelete = (commentId) => {
    const updatedComments = comments.filter((c) => c.id !== commentId);
    setComments(updatedComments);
    setShowCommentPopup(false);
  };
```

Comment (isabeen) [reply components/PairingItem.js]: Ideally, when the user clicks on Delete button, the comments window should close.

**Review by isabeen — CHANGES_REQUESTED** [review]

File: components/PairingItem.js (hunk 1 above)
Comment (Sarahabidar) [reply components/PairingItem.js]: Thank you Sabeen for your suggestion! 
I updated the code to close the comment window when the delete button is clicked.

File: pages/favorites.js
Comment (NikRoe) [inline pages/favorites.js:1]: In your favorites page you are currently unable to see any existing comments for an entry. This file looks only halfway done

File: components/NewCommentForm.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  transition: background 0.3s;
+  &:disabled {
+    background: #ccc;
+    cursor: not-allowed;
+  }
+  &:hover:not(:disabled) {
+    background: #0056b3;
+  }
+`;
+const CommentsList = styled.div`
```
Comment (NikRoe) [inline components/NewCommentForm.js]:
```suggestion
const CommentsList = styled.ul`
```

File: components/NewCommentForm.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  margin-top: 20px;
+`;
+
+const Comment = styled.div`
```
Comment (NikRoe) [inline components/NewCommentForm.js]:
```suggestion
const Comment = styled.li`
```

File: components/NewCommentForm.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+  background: none;
+  border: none;
+  font-size: 24px;
+  cursor: pointer;
+  color: #888;
+  &:hover {
+    color: #000;
+  }
+`;
+const Content = styled.section`
```
Comment (NikRoe) [inline components/NewCommentForm.js]: Make this a label to have basic accessibility:

```suggestion
const Content = styled.label`
```

File: _styles.js [hunk 5]
```diff
@@ -444,46 +447,40 @@ export const SearchResult = styled.li`
...
+  margin-top: 16px;
+  font-size: 14px;
+`;
+
+export const FavoriteButton = styled.button`
+  display: flex;
+  position: relative;
+  margin-left: 85%;
+  margin-top: 15px;
+  background-color: ${({ isFavorite }) => (isFavorite ? "#dc3545" : "#007bff")};
```
Comment (NikRoe) [inline _styles.js]: this should be prefixed with a "$" for styled components, and are you actually using this prop?

```suggestion
  background-color: ${({ $isFavorite }) => ($isFavorite ? "#dc3545" : "#007bff")};
```

File: _styles.js [hunk 6]
```diff
@@ -444,46 +447,40 @@ export const SearchResult = styled.li`
...
+  color: white;
+  border: none;
+  border-radius: 15px;
+  cursor: pointer;
+  font-size: 1rem;
+  transition: background-color 0.3s ease;
+
+  &:hover {
+    background-color: ${({ isFavorite }) =>
+      isFavorite ? "#a71d2a" : "#0056b3"};
```
Comment (NikRoe) [inline _styles.js]:
```suggestion
    background-color: ${({ $isFavorite }) =>
      $isFavorite ? "#a71d2a" : "#0056b3"};
```

File: components/CommentPopup.js [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+  border: none;
+  font-size: 24px;
+  cursor: pointer;
+  color: #888;
+  &:hover {
+    color: #000;
+  }
+`;
+
+const Content = styled.section`
```
Comment (NikRoe) [inline components/CommentPopup.js]:
```suggestion
const Content = styled.label`
```

File: components/PairingItem.js [hunk 8]
```diff
@@ -1,54 +1,76 @@
...
+    <Card>
       <ImageWrapper>
         <StyledImage
           src={pairing.imgUrl}
           alt={pairing.reason}
           fill
           sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw"
+          priority
         />
+        <FavoriteButton onClick={toggleFavoritePairing}>
```
Comment (NikRoe) [inline components/PairingItem.js]: You can currently not toggle favorite pairings. This should fix it:

```suggestion
 <FavoriteButton onClick={() => toggleFavoritePairing(pairing._id)}>
```

File: components/CommentPopup.js [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+  return (
+    <Overlay>
+      <Popup>
+        <Header>
+          <Title>{commentToEdit ? "Edit Comment" : "Comment"}</Title>
+          <CloseButton onClick={onClose}>×</CloseButton>
+        </Header>
+        <Content>
+          <TextArea
+            label="Comment"
```
Comment (NikRoe) [inline components/CommentPopup.js]: This attribute does not exist on a textarea

```suggestion
```

File: components/CommentPopup.js
Comment (NikRoe) [inline components/CommentPopup.js:1]: Where are you using this component?

**Review by NikRoe — CHANGES_REQUESTED** [review]
Hi there,

adding, removing and editing comments for the pairings page work quite nicely 🙌 

However, the favorites page does not work properly. You cannot display comments for pairings since no according logic has been implemented.
There is also a component `CommentPopup` that is not used anywhere in your application.
Currently, you cannot toggle favorite pairings. I left a suggestion on how to fix it.

Please implement the suggestions I left, remove the `CommentPopup` component and implement the needed logic for the favorites page to make it so that comments can also be displayed on the favorites page for pairings.

Keep it up ✌️

**Review by NikRoe — APPROVED** [review]
LGTM

**Pull request merged**